src/extension/survey.ts

```diff
@@ -0,0 +1,150 @@
+import path from 'node:path'
+import { readFileSync, unlinkSync } from 'node:fs'
+
+import {
+  Disposable,
+  NotebookDocument,
+  workspace,
+  window,
+  ExtensionContext,
+  Task,
+  TaskScope,
+  ShellExecution,
+  TaskRevealKind,
+  TaskPanelKind,
+  tasks,
+  Uri,
+  commands,
+} from 'vscode'
+import { TelemetryReporter } from 'vscode-telemetry'
+
+import { Kernel } from './kernel'
+import { isWindows } from './utils'
+
+export class WinDefaultShell implements Disposable {
+  readonly #id: string = 'runme.surveyWinDefaultShell'
+  readonly #tempDir: Uri
+  readonly #context: ExtensionContext
+  readonly #disposables: Disposable[] = []
+
+  constructor(context: ExtensionContext) {
+    commands.registerCommand(this.#id, this.#prompt.bind(this))
+
+    this.#tempDir = context.globalStorageUri
+    this.#context = context
+
+    // Only prompt on Windows
+    if (!isWindows()) {
+      return
+    }
+
+    this.#disposables.push(
+      workspace.onDidOpenNotebookDocument(this.#handleOpenNotebook.bind(this))
+    )
+  }
+
+  async #handleOpenNotebook({ notebookType }: NotebookDocument) {
+    if (
+      notebookType !== Kernel.type ||
+      this.#context.globalState.get<boolean>(
+        this.#id,
+        false
+      )
+    ) {
+      return
+    }
+
+    await new Promise<void>(resolve => setTimeout(resolve, 2000))
+    await commands.executeCommand(this.#id, false)
+  }
+
+  async #prompt(runDirect = true) {
+    // reset done when run from command palette
+    if (runDirect) {
+      await this.#undo()
+    }
+
+    const option = await window.showInformationMessage(
+      'Please help us improve Runme on Windows: Click OK to share what default shell you are using.',
+      'OK',
+      'Don\'t ask again',
+      'Dismiss'
+    )
+    if (option === 'Dismiss') {
+      return
+    } else if (option !== 'OK') {
+      await this.#done()
+      return
+    }
+
```

Comment:
```suggestion
mkdirSync(this.#tempDir.fsPath, { recursive: true })
```

On a fresh install, `this.#tempDir` may not exist, so we should make sure it does

Recursive option here is important